Forge/src/test/groovy/cgltest/Datagen.groovy

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright (C) 2022 GroovyMC and contributors
+ * SPDX-License-Identifier: LGPL-3.0-or-later
+ */
+
+package cgltest
+
+import com.matyrobbrt.gml.bus.EventBusSubscriber
+import com.matyrobbrt.gml.bus.type.ModBus
+import groovy.transform.CompileStatic
+import groovy.transform.PackageScope
+import io.github.groovymc.cgl.api.datagen.recipe.GRecipeProvider
+import io.github.groovymc.cgl.api.datagen.recipe.GShapelessRecipeBuilder
+import net.minecraft.data.PackOutput
+import net.minecraft.data.recipes.RecipeCategory
+import net.minecraft.data.recipes.SimpleCookingRecipeBuilder
+import net.minecraft.resources.ResourceLocation
+import net.minecraft.tags.ItemTags
+import net.minecraft.world.item.Items
+import net.minecraft.world.item.crafting.Ingredient
+import net.minecraftforge.data.event.GatherDataEvent
+import net.minecraftforge.eventbus.api.SubscribeEvent
+
+@CompileStatic
+@EventBusSubscriber(ModBus)
+class Datagen extends GRecipeProvider {
+    @SubscribeEvent @PackageScope
+    static void onDatagen(final GatherDataEvent event) {
+        event.generator.addProvider(event.includeServer(), new Datagen(event.generator.packOutput))
+    }
+
+    Datagen(PackOutput packOutput) {
+        super(packOutput, 'cgltest')
+    }
+
+    @Override
+    protected void buildRecipes() {
+        shaped {
+            result Items.ACACIA_STAIRS count 2
+            category = RecipeCategory.COMBAT
+            group = 'yes'
+
+            pattern 'AAA'
+            define 'A', Items.ACACIA_LOG
+        } save 'my_stairs'
+
+        recipe(GShapelessRecipeBuilder) {
+            result Items.ACACIA_BOAT
+            category = RecipeCategory.REDSTONE
+            group = 'no'
+
+            requires ItemTags.COAL_ORES
+        }.save()
```

Comment:
Are there any scenarios where you wouldn't want to save this immediately? If not, this could be auto-saved.

Reply:
Saving forgotten recipes is already handled by `saveForgotten()`.

Reply:
But are there any cases where you would want to define recipes and not save them? If not, auto-save by default would be more user-friendly.